composer/datasets/ade20k.py

```diff
@@ -0,0 +1,336 @@
+    """
+
+    def __init__(self, ignore_background: bool = True) -> None:
+        self.mean: Optional[Tensor] = None
+        self.std: Optional[Tensor] = None
+        self.ignore_background = ignore_background
+
+    def __call__(self, batch: Batch):
+        xs, ys = batch
+        assert isinstance(xs, Tensor)
+        assert isinstance(ys, Tensor)
+        device = xs.device
+
+        if self.mean is None:
+            self.mean = torch.tensor([0.485 * 255, 0.456 * 255, 0.406 * 255], device=device)
+            self.mean = self.mean.view(1, 3, 1, 1)
+        if self.std is None:
+            self.std = torch.tensor([0.229 * 255, 0.224 * 255, 0.225 * 255], device=device)
+            self.std = self.std.view(1, 3, 1, 1)
+
+        xs = xs.float()
+        xs = xs.sub_(self.mean).div_(self.std)
+        if self.ignore_background:
+            ys = ys.sub_(1)
+        return xs, ys
+
+
+def fast_collate(batch: List[Tuple[Image.Image, Tensor]], memory_format: torch.memory_format = torch.contiguous_format):
+    """Constructs a batch for training from individual samples.
+    """
+    imgs = [sample[0] for sample in batch]
+    targets = [sample[1] for sample in batch]
+    w = imgs[0].size[0]
+    h = imgs[0].size[1]
+    image_tensor = torch.zeros((len(imgs), 3, h, w), dtype=torch.uint8).contiguous(memory_format=memory_format)
+    target_tensor = torch.zeros((len(targets), h, w), dtype=torch.int64).contiguous(memory_format=memory_format)
+    for i, img in enumerate(imgs):
+        nump_array = np.asarray(img, dtype=np.uint8)
+        if nump_array.ndim < 3:
+            nump_array = np.expand_dims(nump_array, axis=-1)
+
+        nump_array = np.rollaxis(nump_array, 2).copy()
+        if nump_array.shape[0] != 3:
+            assert nump_array.shape[0] == 1, "unexpected shape"
+            nump_array = np.resize(nump_array, (3, h, w))
+        assert image_tensor.shape[1:] == nump_array.shape, "shape mismatch"
+
+        image_tensor[i] += torch.from_numpy(nump_array)
+        target_tensor[i] += torch.from_numpy(np.array(targets[i], dtype=np.int64))
+
+    return image_tensor, target_tensor
+
+
+class RandomResizePair(torch.nn.Module):
+    """Randomly select the scale to increase a base size for resizing both the image and target.
+
+    Args:
+        min_scale (float): the minimum value the samples can be rescaled.
+        max_scale (float): the maximum value the samples can be rescaled.
+        base_size (Tuple[int, int]): a specified base size to scale for the resized dimensions.
+    """
+
+    def __init__(self, min_scale: float, max_scale: float, base_size: Optional[Tuple[int, int]] = None):
+        super().__init__()
+        self.min_scale = min_scale
+        self.max_scale = max_scale
+        self.base_size = base_size
+
+    def forward(self, sample: Tuple[Image.Image, Image.Image]):
+        image, target = sample
```

Comment:
I'm a bit confused on how a PIL `Image.Image` would be passed through on the forward pass...wouldn't the inputs be tensors?

Reply:
Ah, I see below that the dataset is calling these functions. Thoughts on having it not inherit from torch.nn.Module and instead rename `forward` to `__call__`?

Reply:
The convention in torchvision is to inherent from torch.nn.Module and use the forward call. I think this is to allow the transforms to be torch scripted. Should we deviate form this standard?

Reply:
Ah makes sense, didn't realize that's how torchvision did it. Let's keep the standard.